boot/bootutil/src/loader.c

```diff
@@ -965,7 +965,8 @@ boot_swap_image(struct boot_loader_state *state, struct boot_status *bs)
 }
 #endif
 
-#if (BOOT_IMAGE_NUMBER > 1)
+#if (BOOT_IMAGE_NUMBER > 1) || \
+    (MCUBOOT_OVERWRITE_ONLY && MCUBOOT_DOWNGRADE_PREVENTION)
```

Comment:
This should be:
```
#if (BOOT_IMAGE_NUMBER > 1) || \
    (defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION))
```
as these are based on whether or not they are defined, not what value they are defined to.